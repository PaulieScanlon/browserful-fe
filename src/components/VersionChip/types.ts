export enum overrideTypes {
  IS_INCLUDED = 'isIncluded',
  IS_EXCLUDED = 'isExcluded'
}
