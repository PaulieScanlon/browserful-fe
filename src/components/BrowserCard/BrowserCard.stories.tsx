import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { State, Store } from '@sambego/storybook-state';

const data2 = require('../../mock-data/data-2.0.modifed.json');

import { BrowserCard } from './BrowserCard';

const store = new Store({
  isExpanded: true
});

const stories = storiesOf('BrowserCard', module);

const onExpandClick = (event, store) => {
  store.set({ isExpanded: !store.get('isExpanded') });
  action('onExpandClick')(
    event.currentTarget,
    `isExpanded: ${store.get('isExpanded')}`
  );
};

const onVersionClick = (event, id) => {
  action('onVersionClick')(event.currentTarget, id);
};

stories.add(
  'default usage',
  withInfo('BrowserCard displays versionChips and a other stuff..')(() => (
    <State store={store}>
      <BrowserCard
        data={data2.agents.edge}
        onExpandClick={event => onExpandClick(event, store)}
        onVersionClick={(event, id) => onVersionClick(event, id)}
      />
    </State>
  ))
);
