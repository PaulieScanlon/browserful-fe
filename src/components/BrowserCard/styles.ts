import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  isExpanded?: boolean;
}

export const BrowserCardWrapper = styled.div({
  label: 'browser-card-wrapper',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1
});

export const BrowserCardHeader = styled.div<IProps>(
  {
    label: 'browser-card-header',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    margin: `0 ${spaceMd}px`,
    padding: `${spaceMd}px 0px`,
    borderBottomWidth: '1px',
    borderBottomStyle: 'solid'
  },
  ({ isExpanded }) => ({
    borderBottomColor: isExpanded ? colours.greyLight : colours.white
  })
);

export const BrowserCardHeaderSection = styled.div({
  label: 'browser-card-header-section',
  display: 'inline-flex',
  alignItems: 'center'
});

export const BrowserCardBody = styled.div<IProps>(
  {
    label: 'browser-card-body',
    overflow: 'hidden'
  },
  ({ isExpanded }) => ({
    height: isExpanded ? '100%' : '0px',
    padding: isExpanded ? `${spaceMd}px ${spaceMd}px` : `0px ${spaceMd}px`
  })
);
