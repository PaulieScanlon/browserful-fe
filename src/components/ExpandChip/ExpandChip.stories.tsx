import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { State, Store } from '@sambego/storybook-state';

import { ExpandChip } from './ExpandChip';

const store = new Store({
  isExpanded: true
});

const stories = storiesOf('ExpandChip', module);

const onClick = (event, store) => {
  store.set({ isExpanded: !store.get('isExpanded') });
  action('onExpandClick')(
    event.currentTarget,
    `isExpanded: ${store.get('isExpanded')}`
  );
};

stories.add(
  'usage',
  withInfo(
    'ExpandChip is used in BrowserCard and contains an onClick to expand the BrowserCard. The isExpaned prop controls the rotation of the chevron'
  )(() => (
    <State store={store}>
      <ExpandChip isExpanded={true} onClick={event => onClick(event, store)} />
    </State>
  ))
);
