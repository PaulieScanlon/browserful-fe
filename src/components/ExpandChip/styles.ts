import styled from 'react-emotion';
import { colours } from '../../theme';

interface IProps {
  isExpaned: boolean;
}

export const ExpandChipWrapper = styled.span<IProps>(
  {
    label: 'expand-chip-wrapper',
    display: 'inline-flex',
    borderRadius: '50%',
    backgroundColor: colours.offWhite,
    cursor: 'pointer',
    svg: {
      fill: colours.greyMid
    }
  },
  ({ isExpaned }) => ({
    transform: isExpaned ? `rotate(0deg)` : `rotate(180deg)`
  })
);
