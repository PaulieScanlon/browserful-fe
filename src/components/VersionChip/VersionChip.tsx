import * as React from 'react';

import {
  VersionLabel,
  VersionText,
  VersionInput,
  VersionStyle
} from './styles';

interface IProps {
  browser: string;
  version: any;
  onChange: (
    event: React.ChangeEventHandler<HTMLInputElement>,
    browser: string,
    version: number | string
  ) => void;
  isIncluded?: boolean;
  defaultChecked?: boolean;
}

export const VersionChip: React.SFC<IProps> = ({
  version,
  browser,
  onChange,
  isIncluded,
  defaultChecked
}: IProps) => {
  return (
    <VersionLabel htmlFor={browser}>
      <VersionInput
        defaultChecked={defaultChecked}
        type="checkbox"
        id={browser}
        onChange={event => onChange(event as any, browser, version)}
        isIncluded={isIncluded}
      />
      <VersionStyle>
        <VersionText isIncluded={isIncluded}>{version || 0}</VersionText>
      </VersionStyle>
    </VersionLabel>
  );
};

VersionChip.defaultProps = {
  isIncluded: false,
  defaultChecked: false
};
