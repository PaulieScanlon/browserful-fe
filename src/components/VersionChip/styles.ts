import styled from 'react-emotion';

import { font } from '../../typography';
import { colours } from '../../theme';

interface IProps {
  isIncluded: boolean;
}

const getStyleRules = (isIncluded, checked) => {
  if (isIncluded) {
    return {
      borderColor: checked ? colours.green : colours.transparent,
      backgroundColor: colours.greenLight
    };
  } else {
    return {
      borderColor: checked ? colours.red : colours.transparent,
      backgroundColor: colours.redLight
    };
  }
};

export const VersionLabel = styled.label({
  label: 'version-label',
  position: 'relative',
  display: 'inline-flex',
  cursor: 'pointer',
  justifyContent: 'center',
  alignItems: 'center'
});

export const VersionText = styled.span<IProps>(
  {
    fontSize: '12px',
    lineHeight: '14px',
    position: 'absolute',
    fontFamily: `${font.fontFamily}`,
    textAlign: 'center',
    whiteSpace: 'pre-wrap',
    textTransform: 'capitalize'
  },
  ({ isIncluded }) => ({
    fontWeight: isIncluded ? 'bold' : 'normal',
    color: isIncluded ? colours.green : colours.red
  })
);

export const VersionInput = styled.input<IProps>(
  {
    label: 'verion-input',
    position: 'absolute',
    display: 'none'
  },
  ({ isIncluded }: IProps) => ({
    ':checked + span': {
      ...getStyleRules(isIncluded, true)
    },
    '+ span': {
      ...getStyleRules(isIncluded, false)
    }
  })
);

export const VersionStyle = styled.span({
  label: 'version-style',
  width: '40px',
  height: '40px',
  borderRadius: '20px',
  borderStyle: 'solid',
  boxSizing: 'border-box',
  outline: 'none'
});
