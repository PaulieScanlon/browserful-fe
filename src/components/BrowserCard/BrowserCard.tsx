import * as React from 'react';

import { browserfulData2 } from '../../types/mainData.types';
import { H6 } from '../../typography';

import {
  BrowserCardWrapper,
  BrowserCardHeader,
  BrowserCardBody,
  BrowserCardHeaderSection
} from './styles';
import { BrowserLogo } from '../BrowserLogo';
import { VersionChip } from '../VersionChip';
import { spaceSm, colours } from '../../theme';
import { Button } from '../Button';

interface IProps {
  data: browserfulData2;
  onExpandClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  onVersionClick: (
    event: React.MouseEvent<HTMLButtonElement>,
    id: string
  ) => void;
  isExpanded?: boolean;
}

export const BrowserCard: React.SFC<IProps> = ({
  data,
  onExpandClick,
  onVersionClick,
  isExpanded
}: IProps) => {
  const { browser, version_list } = data;
  const tempButtonText = isExpanded ? 'Collapse' : 'Expand';

  const versions = version_list.map((obj, i) => {
    return (
      <VersionChip
        key={i}
        browser={browser}
        version={parseInt(obj.version) || 0}
        isIncluded={obj.isIncluded}
        always={obj.always}
        onClick={(event, id) => onVersionClick(event, id)}
      />
    );
  });

  return (
    <BrowserCardWrapper>
      <BrowserCardHeader isExpanded={isExpanded}>
        <BrowserCardHeaderSection>
          <BrowserLogo browser={browser} />
          <H6 marginLeft={`${spaceSm}px`} display="inline-flex">
            {browser}
          </H6>
        </BrowserCardHeaderSection>

        <BrowserCardHeaderSection>
          <Button
            fontColour={colours.greyDark}
            backgroundColour={colours.offWhite}
            onClick={event => onExpandClick(event)}
          >
            {tempButtonText}
          </Button>
        </BrowserCardHeaderSection>
      </BrowserCardHeader>
      <BrowserCardBody isExpanded={isExpanded}>{versions}</BrowserCardBody>
    </BrowserCardWrapper>
  );
};

BrowserCard.defaultProps = {
  isExpanded: true,
  onExpandClick: () => {},
  onVersionClick: () => {}
};
