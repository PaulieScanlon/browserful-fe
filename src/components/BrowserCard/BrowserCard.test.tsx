import * as React from 'react';
import { shallow } from 'enzyme';

import { coverToBrowserfulData2 } from '../../modules/browserful-data-2.0';

const data2Subset = require('../../mock-data/data-2.0.subset.json');

import { BrowserCard } from './BrowserCard';
import { ExpandChip } from '../ExpandChip';
import { VersionChip } from '../VersionChip';

const onExpandClick = event => {};
const onVersionClick = (event, id) => {};

const shallowDefault = () => {
  return shallow(
    <BrowserCard
      data={coverToBrowserfulData2(data2Subset).agents.edge}
      onExpandClick={event => onExpandClick(event)}
      onVersionClick={(event, id) => onVersionClick(event, id)}
    />
  );
};

describe('<BrowserCard />', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault();
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls expand function on click', () => {
    const testOnExpandClick = jest.fn();
    const wrapper = shallow(
      <BrowserCard
        data={coverToBrowserfulData2(data2Subset).agents.edge}
        onExpandClick={event => testOnExpandClick(event)}
      />
    );
    const expandChip = wrapper.find(ExpandChip);
    expandChip.simulate('click');
    expect(testOnExpandClick).toHaveBeenCalledTimes(1);
  });
  it('calls version function on click', () => {
    const testVersionOnClick = jest.fn();
    const wrapper = shallow(
      <BrowserCard
        data={coverToBrowserfulData2(data2Subset).agents.edge}
        onVersionClick={(event, id) => testVersionOnClick(event, id)}
      />
    );
    const versionChip = wrapper.find(VersionChip).at(0);
    versionChip.simulate('click');
    expect(testVersionOnClick).toHaveBeenCalledTimes(1);
  });
});
