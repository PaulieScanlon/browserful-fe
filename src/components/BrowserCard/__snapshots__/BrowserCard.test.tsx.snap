// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowserCard /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserCard
    data={
      Object {
        "abbr": "Edge",
        "browser": "Edge",
        "current_version": "17",
        "isExpanded": true,
        "prefix": "ms",
        "totalVersions": 7,
        "type": "desktop",
        "usage_global": Object {
          "12": 0.02163,
          "13": 0.025956,
          "14": 0.056238,
          "15": 0.06489,
          "16": 0.19467,
          "17": 1.47949,
          "18": 0,
        },
        "version_list": Array [
          Object {
            "always": false,
            "era": -5,
            "global_usage": 0.02163,
            "isIncluded": false,
            "prefix": "",
            "release_date": 1438128000,
            "version": "12",
          },
          Object {
            "always": false,
            "era": -4,
            "global_usage": 0.025956,
            "isIncluded": false,
            "prefix": "",
            "release_date": 1447286400,
            "version": "13",
          },
          Object {
            "always": false,
            "era": -3,
            "global_usage": 0.056238,
            "isIncluded": false,
            "prefix": "",
            "release_date": 1470096000,
            "version": "14",
          },
          Object {
            "always": false,
            "era": -2,
            "global_usage": 0.06489,
            "isIncluded": false,
            "prefix": "",
            "release_date": 1491868800,
            "version": "15",
          },
          Object {
            "always": false,
            "era": -1,
            "global_usage": 0.19467,
            "isIncluded": false,
            "prefix": "",
            "release_date": 1508198400,
            "version": "16",
          },
          Object {
            "always": false,
            "era": 0,
            "global_usage": 1.47949,
            "isIncluded": false,
            "prefix": "",
            "release_date": 1525046400,
            "version": "17",
          },
          Object {
            "always": false,
            "era": 1,
            "global_usage": 0,
            "isIncluded": false,
            "prefix": "",
            "release_date": null,
            "version": "18",
          },
        ],
      }
    }
    isExpanded={true}
    onExpandClick={[Function]}
    onVersionClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BrowserCardHeader
          isExpanded={true}
        >
          <BrowserCardHeaderSection>
            <BrowserLogo
              browser="Edge"
            />
            <H6
              display="inline-flex"
              marginLeft="5px"
            >
              Edge
            </H6>
          </BrowserCardHeaderSection>
          <BrowserCardHeaderSection>
            <ExpandChip
              isExpanded={true}
              onClick={[Function]}
            />
          </BrowserCardHeaderSection>
        </BrowserCardHeader>,
        <BrowserCardBody
          isExpanded={true}
        >
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={12}
          />
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={13}
          />
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={14}
          />
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={15}
          />
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={16}
          />
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={17}
          />
          <VersionChip
            always={false}
            browser="Edge"
            isIncluded={false}
            onClick={[Function]}
            version={18}
          />
          <BrowserCardFooter>
            BrowserCardFooter
          </BrowserCardFooter>
        </BrowserCardBody>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <BrowserCardHeaderSection>
              <BrowserLogo
                browser="Edge"
              />
              <H6
                display="inline-flex"
                marginLeft="5px"
              >
                Edge
              </H6>
            </BrowserCardHeaderSection>,
            <BrowserCardHeaderSection>
              <ExpandChip
                isExpanded={true}
                onClick={[Function]}
              />
            </BrowserCardHeaderSection>,
          ],
          "isExpanded": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <BrowserLogo
                  browser="Edge"
                />,
                <H6
                  display="inline-flex"
                  marginLeft="5px"
                >
                  Edge
                </H6>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "browser": "Edge",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Edge",
                  "display": "inline-flex",
                  "marginLeft": "5px",
                },
                "ref": null,
                "rendered": "Edge",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ExpandChip
                isExpanded={true}
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "isExpanded": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            Array [
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={12}
              />,
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={13}
              />,
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={14}
              />,
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={15}
              />,
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={16}
              />,
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={17}
              />,
              <VersionChip
                always={false}
                browser="Edge"
                isIncluded={false}
                onClick={[Function]}
                version={18}
              />,
            ],
            <BrowserCardFooter>
              BrowserCardFooter
            </BrowserCardFooter>,
          ],
          "isExpanded": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 12,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 13,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 14,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 15,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 17,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "function",
            "props": Object {
              "always": false,
              "browser": "Edge",
              "isIncluded": false,
              "onClick": [Function],
              "version": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "BrowserCardFooter",
            },
            "ref": null,
            "rendered": "BrowserCardFooter",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BrowserCardHeader
            isExpanded={true}
          >
            <BrowserCardHeaderSection>
              <BrowserLogo
                browser="Edge"
              />
              <H6
                display="inline-flex"
                marginLeft="5px"
              >
                Edge
              </H6>
            </BrowserCardHeaderSection>
            <BrowserCardHeaderSection>
              <ExpandChip
                isExpanded={true}
                onClick={[Function]}
              />
            </BrowserCardHeaderSection>
          </BrowserCardHeader>,
          <BrowserCardBody
            isExpanded={true}
          >
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={12}
            />
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={13}
            />
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={14}
            />
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={15}
            />
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={16}
            />
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={17}
            />
            <VersionChip
              always={false}
              browser="Edge"
              isIncluded={false}
              onClick={[Function]}
              version={18}
            />
            <BrowserCardFooter>
              BrowserCardFooter
            </BrowserCardFooter>
          </BrowserCardBody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <BrowserCardHeaderSection>
                <BrowserLogo
                  browser="Edge"
                />
                <H6
                  display="inline-flex"
                  marginLeft="5px"
                >
                  Edge
                </H6>
              </BrowserCardHeaderSection>,
              <BrowserCardHeaderSection>
                <ExpandChip
                  isExpanded={true}
                  onClick={[Function]}
                />
              </BrowserCardHeaderSection>,
            ],
            "isExpanded": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <BrowserLogo
                    browser="Edge"
                  />,
                  <H6
                    display="inline-flex"
                    marginLeft="5px"
                  >
                    Edge
                  </H6>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "browser": "Edge",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Edge",
                    "display": "inline-flex",
                    "marginLeft": "5px",
                  },
                  "ref": null,
                  "rendered": "Edge",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ExpandChip
                  isExpanded={true}
                  onClick={[Function]}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "isExpanded": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              Array [
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={12}
                />,
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={13}
                />,
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={14}
                />,
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={15}
                />,
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={16}
                />,
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={17}
                />,
                <VersionChip
                  always={false}
                  browser="Edge"
                  isIncluded={false}
                  onClick={[Function]}
                  version={18}
                />,
              ],
              <BrowserCardFooter>
                BrowserCardFooter
              </BrowserCardFooter>,
            ],
            "isExpanded": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 12,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 13,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 14,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 15,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 16,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 17,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "function",
              "props": Object {
                "always": false,
                "browser": "Edge",
                "isIncluded": false,
                "onClick": [Function],
                "version": 18,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "BrowserCardFooter",
              },
              "ref": null,
              "rendered": "BrowserCardFooter",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
